Index: content/browser/zygote_host_linux.cc
===================================================================
--- content/browser/zygote_host_linux.cc	(revision 108895)
+++ content/browser/zygote_host_linux.cc	(working copy)
@@ -28,6 +28,7 @@
 #include "content/common/process_watcher.h"
 #include "content/common/result_codes.h"
 #include "content/common/unix_domain_socket_posix.h"
+#include "content/common/child_process_host.h"
 #include "sandbox/linux/suid/suid_unsafe_environment_variables.h"
 
 #if defined(USE_TCMALLOC)
@@ -78,8 +79,7 @@
   DCHECK(!init_);
   init_ = true;
 
-  FilePath chrome_path;
-  CHECK(PathService::Get(base::FILE_EXE, &chrome_path));
+  FilePath chrome_path = ChildProcessHost::GetChildPath(ChildProcessHost::CHILD_NORMAL);
   CommandLine cmd_line(chrome_path);
 
   cmd_line.AppendSwitchASCII(switches::kProcessType, switches::kZygoteProcess);
