Index: base/message_pump_mac.mm
===================================================================
--- base/message_pump_mac.mm	(revision 76624)
+++ base/message_pump_mac.mm	(working copy)
@@ -714,10 +714,7 @@
 // returning it to its previous value once the event has been sent.
 NSAutoreleasePool* MessagePumpNSApplication::CreateAutoreleasePool() {
   NSAutoreleasePool* pool = nil;
-  DCHECK([NSApp conformsToProtocol:@protocol(CrAppProtocol)]);
-  if (![NSApp isHandlingSendEvent]) {
-    pool = MessagePumpCFRunLoopBase::CreateAutoreleasePool();
-  }
+  pool = MessagePumpCFRunLoopBase::CreateAutoreleasePool();
   return pool;
 }
 
